tests/common/test_logger.py

```diff
@@ -161,34 +174,336 @@ def test_periodic_forwards_message_and_args(self, mock_info, mock_error, mock_wa
+        logger.add_logger_appender(logger.AppenderType.TELEMETRY, logger.LogLevel.WARNING, path=add_log_event)
+        logger.add_logger_appender(logger.AppenderType.CONSOLE, logger.LogLevel.WARNING, path="/dev/null")
+        logger.add_logger_appender(logger.AppenderType.STDOUT, logger.LogLevel.WARNING)
+        logger.set_prefix(parent_prefix)
+
+        lg = logger.Logger(logger.DEFAULT_LOGGER, child_prefix)
+
+        lg.error("Test Log")
+        self.assertEqual(1, mock_file_write.call_count)
+        self.assertEqual(1, mock_console_write.call_count)
+        self.assertEqual(1, mock_telem_write.call_count)
+        self.assertEqual(1, mock_stdout_write.call_count)
+
+        self.assertIn(child_prefix, mock_file_write.call_args[0][1])
+        self.assertIn(child_prefix, mock_console_write.call_args[0][1])
+        self.assertIn(child_prefix, mock_telem_write.call_args[0][1])
+        self.assertIn(child_prefix, mock_stdout_write.call_args[0][1])
+
+        logger.error("Test Log")
+        self.assertEqual(2, mock_file_write.call_count)
+        self.assertEqual(2, mock_console_write.call_count)
+        self.assertEqual(2, mock_telem_write.call_count)
+        self.assertEqual(2, mock_stdout_write.call_count)
+
+        self.assertIn(parent_prefix, mock_file_write.call_args[0][1])
+        self.assertIn(parent_prefix, mock_console_write.call_args[0][1])
+        self.assertIn(parent_prefix, mock_telem_write.call_args[0][1])
+        self.assertIn(parent_prefix, mock_stdout_write.call_args[0][1])
+
+    @patch("azurelinuxagent.common.event.send_logs_to_telemetry", return_value=True)
+    @patch('azurelinuxagent.common.logger.Logger.error')
+    @patch('azurelinuxagent.common.logger.Logger.warn')
+    def test_telemetry_logger_verify_not_logging_errors_warnings(self, mock_warn, mock_error, *args):
+        appender = logger.TelemetryAppender(logger.LogLevel.WARNING, add_log_event)
+
+        with patch('azurelinuxagent.common.event.EventLogger.save_event') as mock_save:
+            appender.write(logger.LogLevel.WARNING, 'Cgroup controller "memory" is not mounted. '
+                                                    'Microsoft.OSTCExtensions.DummyExtension-1.2.3.4')
+            self.assertEqual(1, mock_save.call_count)
+            self.assertEqual(0, mock_warn.call_count)
+            self.assertEqual(0, mock_error.call_count)
+
+        # Writing 2000 events should generate only one more log event due to too many files. #1035 was caused due to
+        # too many files being written in an error condition (earlier code would write 1000 files in this case).
+        for _ in range(2000):
+            appender.write(logger.LogLevel.WARNING, 'Cgroup controller "memory" is not mounted. '
+                                                    'Microsoft.OSTCExtensions.DummyExtension-1.2.3.4')
+
+        self.assertEqual(1, mock_warn.call_count)
```

Comment:
maybe we need to rename the test: I read "test_telemetry_logger_verify_not_logging_errors_warnings" as implying that no warnings/errors should be logged, but we do log a warning

Reply:
Same as above - Discussed offline - Removing this test as it is not relevant now. The purpose of this test was to make sure no additional logs are added, especially with #1035-like issue. But with the new fix for #1035 done, it is not needed anymore.